bip-xxxx.mediawiki

```diff
@@ -0,0 +1,338 @@
+}
+</pre>
+
+The well-known error codes are:
+{| class="wikitable"
+!Error code
+!Meaning
+|-
+|leaking-data
+|Key path information or GlobalXPubs should not be included in the original PSBT.
+|-
+|psbt-not-finalized
+|The original PSBT must be finalized.
+|-
+|unavailable
+|The payjoin endpoint is not available for now.
+|-
+|out-of-utxos
+|The receiver does not have any UTXO to contribute in a payjoin proposal.
+|-
+|not-enough-money
+|The receiver added some inputs but could not bump the fee of the payjoin proposal.
+|-
+|insane-psbt
+|Some consistency check on the PSBT failed.
+|-
+|version-unsupported
+|This version of payjoin is not supported.
+|-
+|need-utxo-information
+|The witness UTXO or non witness UTXO is missing
+|-
+|invalid-transaction
+|The original transaction is invalid for payjoin
+|}
+
+The receiver is allowed to return implementation specific errors which may assist the sender to diagnose any issue.
+
+However, it is important that error codes that are not well-known and that the message do not appear on the sender's software user interface.
+Such error codes or messages could be used maliciously to phish a non technical user.
+Instead those errors or messages can only appear in debug logs.
+
+It is advised to hard code the description of the error codes into the sender's software.
+
+===Receiver's original PSBT checklist===
+
+The receiver needs to do some check on the original PSBT before proceeding:
+
+* Non-interactive receivers (like a payment processor) need to check that the original PSBT is broadcastable. <code>*</code>
+* If the sender included inputs in the original PSBT owned by the receiver, the receiver must either return error <code>invalid-transaction</code> or make sure they do not sign those inputs in the payjoin proposal.
+* If the sender's inputs are all from the same scriptPubKey type, the receiver must match the same type. If the receiver can't match the type, they must return error <code>out-of-utxos</code>.
+
+<code>*</code>: Interactive receivers are not required to validate the original PSBT because they are not exposed to probing attacks.
+
+===Sender's payjoin proposal checklist===
+
+The sender should check the payjoin proposal before signing it to prevent a malicious receiver from stealing money.
+
+* Check that all the spent outpoints in the original PSBT still exist in the coinjoin PSBT.
+* Check that all the spent outpoints in the original PSBT do not have any partial signature.
+* If the sender is not using inputs with mixed types, check that the receiver inputs type match the inputs type of the sender. (ie. both using P2SH-P2WPKH or both using P2WPKH)
+* Check that any inputs added by the receiver are finalized.
+* Check that the transaction version, and nLockTime are unchanged.
+* Check that the sender's inputs' sequence numbers are unchanged.
+* If the sender's inputs' sequence numbers the homogenous, check that the receiver's contributed inputs match those.
```

Comment:
s/the homogeneous/are homogeneous/ but to be honest I think "homogeneous" is an unnecessarily fancy word choice here, to be ultra clear I prefer "If the sender's inputs' sequence numbers are all the same, ..."